src/qibocal/fitting/methods.py

```diff
@@ -1073,3 +1078,83 @@ def calibrate_qubit_states_fit(data, x, y, nshots, qubits, degree=True):
         }
         parameters.add(results)
     return parameters
+
+
+def punchout_fit(data, qubits, resonator_type, labels, fit_type):
+    """Fit frequency and attenuation at high and low power for a given resonator
+        Args:
+        data (DataUnits): data file with information on the feature response at each current point.
+        qubits (list): qubits coupled to the resonator that we are probing.
+        resonator_type (str): the type of readout resonator ['3D', '2D'].
+        labels (list of str): list containing the lables of the quantities computed by this fitting method.
+        fit_type (str): the type of punchout executed ['attenuation', 'amplitude'].
+
+    Returns:
+        data_fit (Data): Data file with labels and fit parameters (frequency at low and high power, attenuation range for low and highg power)
+    """
+
+    data_fit = Data(
+        name=f"fits",
+        quantities=[
+            labels[0],
+            labels[1],
+            labels[2],
+            labels[3],
+            labels[4],
+            labels[5],
+            "qubit",
+        ],
+    )
+
+    for qubit in qubits:
+        averaged_data = (
+            data.df[data.df["qubit"] == qubit]
+            .drop(columns=["i", "q", "qubit", "iteration"])
+            .groupby(["frequency", fit_type], as_index=False)
+            .mean()
+        )
+        try:
+            normalised_data = averaged_data.groupby([fit_type], as_index=False)[
+                ["MSR"]
+            ].transform(norm)
+
+            averaged_data_updated = averaged_data.copy()
+            averaged_data_updated.update(normalised_data["MSR"])
+
+            min_points = find_min_msr(averaged_data_updated, resonator_type, fit_type)
+            min_points = [[q.magnitude for q in tupla] for tupla in min_points]
+            x = [point[0] for point in min_points]
+            y = [point[1] for point in min_points]
```

Comment:
if `min_points` is a np matrix you can use this syntax
```suggestion
            x = min_points[: ,0]
            y = min_points[:, 1]
```